src/replace.rs

```diff
@@ -36,7 +37,7 @@ impl Data {
                 Span {
                     data: State::Initial,
                     start: 0,
-                    end: data.len(),
+                    end: data.len() - 1,
```

Comment:
Let's replace this with `data.len().saturating_sub(1)` so it can never underflow (`0.saturating_sub(1) == 0`)